src/Aspire.Dashboard/Components/ResourcesGridColumns/ResourceNameDisplay.razor

```diff
@@ -3,17 +3,37 @@
 @using Aspire.Dashboard.Model
 @using Aspire.Dashboard.Otlp.Model
 @using Aspire.Dashboard.Resources
+
 @inject IStringLocalizer<Columns> Loc
 
+@if (Resource.Properties.TryGetValue(KnownProperties.Container.Lifetime, out var value) &&
+    value.Value.HasStringValue &&
+    StringComparer.Ordinal.Equals(value.Value.StringValue, KnownContainerLifetimes.Persistent))
+{
+    <FluentIcon
+        Icon="Icons.Regular.Size16.Shield"
+        Title="@Loc[nameof(Columns.PersistentContainerIconTooltip)]"
+        Color="Color.Neutral"
+        Class="severity-icon" />
+}
+
 <span title="@FormatName(Resource)"><FluentHighlighter HighlightedText="@FilterText" Text="@FormatName(Resource)"/></span>
 
 @code {
     [Parameter, EditorRequired]
-    public required ResourceViewModel Resource { get; set; }
+    public required ResourceViewModel Resource { get; init; }
 
     [Parameter, EditorRequired]
-    public required Func<ResourceViewModel, string> FormatName { get; set; }
+    public required Func<ResourceViewModel, string> FormatName { get; init; }
 
     [Parameter, EditorRequired]
-    public required string FilterText { get; set; }
+    public required string FilterText { get; init; }
+
+    private static class KnownContainerLifetimes
+    {
+        // Matches members from Aspire.Hosting.ApplicationModel.ContainerLifetime
+
+        public const string Default = nameof(Default);
```

Comment:
If a lifetime isn't set to persistent the UI will show Default. Do we think that is discriptive enough? I could imagine people seeing the value and no knowing what default means.

Maybe show "Session" or "Transient" as the lifetime instead of default.

Reply:
Agree that "Default" isn't super helpful. I don't love hardcoding the intepretation of the default value into the dashboard, but we could to the translation in the app host. Either way, the value isn't localised.

Another option is to change the property to boolean `IsPersistent`.

Reply:
I don't think we have any true/false properties.

Maybe `ContainerLifetime` enum itself should change from Default to Session? @mitchdenny @davidfowl

Reply:
I started a teams discussion. Doesn't need to block this PR.